### PR TITLE
build: remove babel

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Remove `babel` and use `ts-node` / `ts-jest`. 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -8,8 +8,7 @@
       "program": "${workspaceFolder}/lib/renovate.ts",
       "env": { "LOG_LEVEL": "debug" },
       "console": "integratedTerminal",
-      "runtimeExecutable": "${workspaceFolder}/node_modules/.bin/babel-node",
-      "runtimeArgs": ["--extensions", ".ts,.js"],
+      "runtimeArgs": ["-r", "ts-node/register"],
```
Comment (viceice) [inline .vscode/launch.json:11]:
```suggestion
      "runtimeArgs": ["-r", "ts-node/register/transpile-only"],
```
Would be a litle more performant

File: jest.config.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
+import type { InitialOptionsTsJest } from 'ts-jest/dist/types';
```
Comment (viceice) [inline jest.config.ts:1]: thanks to ts-node this can now be typescript too 🙃

File: jest.config.ts [hunk 3]
```diff
@@ -25,7 +28,11 @@ module.exports = {
   snapshotSerializers: ['<rootDir>/test/newline-snapshot-serializer.ts'],
   testEnvironment: 'node',
   testRunner: 'jest-circus/runner',
-  transform: {
-    '^.+\\.(j|t)s$': 'babel-jest',
+  globals: {
+    'ts-jest': {
+      diagnostics: false,
```
Comment (viceice) [inline jest.config.ts:33]: only for speed

File: lib/config/file.spec.ts [hunk 4]
```diff
@@ -29,10 +40,7 @@ describe('config/file', () => {
       expect(res.rangeStrategy).toEqual('bump');
     });
     it('informs user when error in parsing config.js', () => {
-      const configFile = upath.resolve(
-        __dirname,
-        './config/__fixtures__/file3.ts'
-      );
+      const configFile = upath.resolve(tmp.path, './file3.js');
```
Comment (viceice) [inline lib/config/file.spec.ts:43]: Completely wrong tests in this file

File: lib/manager/index.spec.ts [hunk 5]
```diff
@@ -27,6 +27,9 @@ describe('manager', () => {
       if (!module.extractPackageFile && !module.extractAllPackageFiles) {
         return false;
       }
+      if (Object.values(module).some((v) => v === undefined)) {
+        return false;
+      }
```
Comment (viceice) [inline lib/manager/index.spec.ts:32]: coverage hack 🙃 So reference all exported properties

File: lib/platform/bitbucket-server/types.ts [hunk 6]
```diff
@@ -19,11 +19,7 @@ export interface BbsPr extends Pr {
   version?: number;
 }
 
-export enum BbsRestPrState {
-  Declined = 'DECLINED',
-  Open = 'OPEN',
-  Merged = 'MERGED',
-}
+export type BbsRestPrState = 'DECLINED' | 'OPEN' | 'MERGED';
```
Comment (viceice) [inline lib/platform/bitbucket-server/types.ts:22]: Coverage fix, we only use this for typings, so no need for an enum

File: lib/platform/bitbucket/utils.spec.ts [hunk 7]
```diff
@@ -35,6 +40,11 @@ describe('accumulateValues()', () => {
     expect(res).toHaveLength(25);
     expect(httpMock.getTrace()).toHaveLength(3);
     expect(httpMock.getTrace()).toMatchSnapshot();
-    httpMock.reset();
+  });
+
+  it('isConflicted', () => {
+    expect(
+      utils.isConflicted([{ chunks: [{ changes: [{ content: '+=======' }] }] }])
+    ).toBe(true);
```
Comment (viceice) [inline lib/platform/bitbucket/utils.spec.ts:48]: Coverage

File: lib/util/html.spec.ts [hunk 8]
```diff
@@ -8,6 +8,7 @@ describe(getName(__filename), () => {
     const div = body.childNodes[0] as HTMLElement;
     expect(div.tagName).toBe('DIV');
     expect(div.textContent).toBe('Hello, world!');
+    expect(div instanceof HTMLElement).toBe(true);
```
Comment (viceice) [inline lib/util/html.spec.ts:11]: coverage

File: package.json [hunk 9]
```diff
@@ -12,11 +12,11 @@
...
     "compile:ts": "tsc -p tsconfig.app.json",
     "compile:dts": "tsc -p tsconfig.dts.json",
-    "config-validator": "babel-node --extensions \".ts,.js\" -- lib/config-validator.ts",
+    "config-validator": "node -r ts-node/register/transpile-only -- lib/config-validator.ts",
     "generate": "run-s generate:*",
     "generate:imports": "node tmp/tools/generate-imports.js",
-    "create-json-schema": "babel-node --extensions \".ts,.js\" -- bin/create-json-schema.js && prettier --write \"renovate-schema.json\"",
-    "debug": "babel-node --inspect-brk  --extensions \".ts,.js\" -- lib/renovate.ts",
+    "create-json-schema": "node -r ts-node/register/transpile-only -- bin/create-json-schema.js && prettier --write \"renovate-schema.json\"",
+    "debug": "node --inspect-brk -r ts-node/register/transpile-only  -- lib/renovate.ts",
```
Comment (viceice) [inline package.json:19]: using register allows additional node args

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
e2e test

image

**Review by rarkins — APPROVED** [review]

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**